src/guardian/guardian.service.ts

```diff
@@ -136,14 +137,25 @@ export class GuardianService implements OnModuleInit {
         blockHash: blockData.blockHash,
       });
 
-      await Promise.all(
-        stakingModules.map(async (stakingRouterModule) => {
-          const stakingModuleData =
-            await this.stakingModuleGuardService.getStakingRouterModuleData(
-              stakingRouterModule,
-              blockHash,
-            );
+      // TODO: check only if one of nonce changed
+      const modulesIdWithDuplicateKeys: number[] =
+        this.stakingModuleGuardService.checkVettedKeysDuplicates(
+          stakingModulesData,
+          blockData,
+        );
+
+      const stakingModulesWithoutDuplicates: StakingModuleData[] =
```

Comment:
I would advice to rewrite this function to
-> Fetch all modules and vetted unused keys for it
-> for module in modules
    -> Promiss all
        -> Check duplicates
        -> Check keys validity
        -> Send deposit message

Reply:
Currently function with name `checkKeysIntersections` somewhy send deposit message, but should only check intersections.

Reply:
i dont think your algorithm is right
we check duplicated keys across all modules and exclude modules with duplicates

Reply:
and you skip intersection check as well
Promis.all also does make sense to put inside for module in modules

Reply:
checkKeysIntersections did it before, this pr didnt change it
   refactoring of council daemon could be done anytime later